tests/agents/test_interactive_textual.py

```diff
@@ -398,9 +398,13 @@ async def test_whitelist_actions_bypass_confirmation():
     )
 
     async with app.run_test() as pilot:
-        # Start the agent with the task
+        await pilot.pause(0.2)  # Wait for UI to be ready
         threading.Thread(target=lambda: app.agent.run("Whitelist test"), daemon=True).start()
-        await pilot.pause(0.2)
+
+        for _ in range(50):
```

Comment:
5s is maybe a bit excessive

```suggestion
        for _ in range(10):
```